File: mapir-android-service-sdk/src/main/java/ir/map/servicesdk/BitmapWorkerTask.java
package ir.map.servicesdk;

import android.graphics.BitmapFactory;
import android.os.AsyncTask;

import java.io.InputStream;

import ir.map.servicesdk.response.StaticMapResponse;

public class BitmapWorkerTask extends AsyncTask<InputStream, Void, StaticMapResponse> {

    private ResponseListener<StaticMapResponse> listener;

    BitmapWorkerTask(ResponseListener<StaticMapResponse> listener) {
        this.listener = listener;
    }

    @Override
    protected StaticMapResponse doInBackground(InputStream... inputStreams) {
        return new StaticMapResponse(inputStreams[0], BitmapFactory.decodeStream(inputStreams[0]));
    }

    @Override
    protected void onPostExecute(StaticMapResponse response) {
        super.onPostExecute(response);

        listener.onSuccess(response);
    }
}

File: mapir-android-service-sdk-ktx/src/main/java/ir/map/servicesdk/BitmapWorkerTask.kt
package ir.map.servicesdk

import android.graphics.BitmapFactory
import android.os.AsyncTask
import ir.map.servicesdk.response.StaticMapResponse
import java.io.InputStream

internal class BitmapWorkerTask (private val listener: ResponseListener<StaticMapResponse>) : AsyncTask<InputStream?, Void?, StaticMapResponse?>() {
    override fun doInBackground(vararg inputStreams: InputStream?): StaticMapResponse? {
        return inputStreams[0]?.let { StaticMapResponse(it, BitmapFactory.decodeStream(inputStreams[0])) }
    }

    override fun onPostExecute(response: StaticMapResponse?) {
        super.onPostExecute(response)

        response?.let { listener.onSuccess(it) }
    }

}